drivers/pinctrl/renesas/ra/pinctrl_renesas_ra8.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/pinctrl.h>
+#include <soc.h>
+
+int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintptr_t reg)
+{
+	R_PMISC->PWPRS = 0;
+	R_PMISC->PWPRS = BIT(R_PMISC_PWPR_PFSWE_Pos);
+
+	for (uint8_t i = 0U; i < pin_cnt; i++) {
+		const pinctrl_soc_pin_t *pin = &pins[i];
+
+		/* PMR bits should be cleared before specifying PSEL. Reference section "20.7 Notes
+		 * on the PmnPFS Register Setting" in the RA6M3 manual R01UH0886EJ0100.
+		 */
+		if ((pin->cfg & BIT(R_PFS_PORT_PIN_PmnPFS_PMR_Pos)) > 0) {
+			/* Clear PMR */
+			R_PFS->PORT[pin->port_num].PIN[pin->pin_num].PmnPFS_b.PMR = 0;
```

Comment:
There are various ways to do this, but
`R_PFS` is a macro, which makes debugging difficult.
The Zephyr-friendly method is to reference registers defined in the device tree via config, as is done with GPIO.

https://github.com/zephyrproject-rtos/zephyr/pull/72150/files#diff-4fd9e45da58f81a8a9761aefd7ed42633287f99d1ad17a6b230bf0fdfa73681dR176

As pointed out above, using the BSP functions is also an option.

Either way, it's best to make your policy clear.
In my opinion, I recommend the Zephyr method, which uses a device tree.